runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -281,17 +305,37 @@ void doSerializeInternal(ByteBuf buf) {
                 int size = buf.writerIndex() - (lengthIndex + 4);
                 buf.writerIndex(lengthIndex);
                 buf.writeInt(size);
+                //System.out.println("Size:  " + size);
                 buf.writerIndex(lengthIndex + size + 4);
-                lastKnownSize = size;
             } else {
+                //System.out.println("Data Len: " + data.length);
                 ICorfuPayload.serialize(buf, data);
-                lastKnownSize = data.length;
             }
         }
-
         if (type.isMetadataAware()) {
-            ICorfuPayload.serialize(buf, metadataMap);
+            if (serializedMetadata == null) {
+                int lengthIndex = buf.writerIndex();
+                buf.writeInt(0);
```

Comment:
Why does it need write a dummy Int?

Reply:
If we look at how the payload is serialized in this function, it writes the size of payload followed by bytes.  This is to keep parity with it.